systest/backup/nfs-backup/backup_test.go

```diff
@@ -0,0 +1,326 @@
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/dgraph-io/dgraph/systest/backup/common"
+	"github.com/dgraph-io/dgraph/testutil"
+	"github.com/dgraph-io/dgraph/worker"
+	"github.com/dgraph-io/dgraph/x"
+
+	"github.com/dgraph-io/dgo/v210"
+	"github.com/dgraph-io/dgo/v210/protos/api"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+)
+
+var (
+	copyBackupDir  = "./data/backups_copy"
+	backupDir      = "./data/backup"
+	restoreDir     = "./data/restore"
+	testDirs       = []string{backupDir, restoreDir}
+	backupDstHA    = "/ha_backup"
+	backupDstNonHA = "/non_ha_backup"
+)
+
+func TestBackupHAClust(t *testing.T) {
+
+	backupRestoreTest(t, testutil.SockAddr, testutil.SockAddrAlpha4Http, testutil.SockAddrZeroHttp, backupDstHA, testutil.SockAddrHttp)
+}
+
+func TestBackupNonHAClust(t *testing.T) {
+
+	backupRestoreTest(t, testutil.SockAddrAlpha7, testutil.SockAddrAlpha8Http, testutil.SockAddrZero7Http, backupDstNonHA, testutil.SockAddrAlpha7Http)
+}
+
+func backupRestoreTest(t *testing.T, backupAlphaSocketAddr string, restoreAlphaAddr string, backupZeroAddr string, backupDst string, backupAlphaSocketAddrHttp string) {
+	conn, err := grpc.Dial(backupAlphaSocketAddr, grpc.WithInsecure())
+	require.NoError(t, err)
+	dg := dgo.NewDgraphClient(api.NewDgraphClient(conn))
+	ctx := context.Background()
+	require.NoError(t, dg.Alter(ctx, &api.Operation{DropAll: true}))
+	// Add schema and types.
+	require.NoError(t, dg.Alter(ctx, &api.Operation{Schema: `movie: string .
+		 type Node {
+			 movie
+		 }`}))
+
+	// Add initial data.
+	original, err := dg.NewTxn().Mutate(ctx, &api.Mutation{
+		CommitNow: true,
+		SetNquads: []byte(`
+			 <_:x1> <movie> "BIRDS MAN OR (THE UNEXPECTED VIRTUE OF IGNORANCE)" .
+			 <_:x2> <movie> "Spotlight" .
+			 <_:x3> <movie> "Moonlight" .
+			 <_:x4> <movie> "THE SHAPE OF WATERLOO" .
+			 <_:x5> <movie> "BLACK PUNTER" .
+		 `),
+	})
+	require.NoError(t, err)
+	t.Logf("--- Original uid mapping: %+v\n", original.Uids)
+	// Move tablet to group 1 to avoid messes later.
+	// client := *http.DefaultClient
+	_, err = http.Get("http://" + backupZeroAddr + "/moveTablet?tablet=movie&group=1")
+	require.NoError(t, err)
+	// After the move, we need to pause a bit to give zero a chance to quorum.
+	t.Log("Pausing to let zero move tablet...")
+	moveOk := false
+	for retry := 5; retry > 0; retry-- {
+		state, err := testutil.GetState()
+		require.NoError(t, err)
+		if _, ok := state.Groups["1"].Tablets[x.NamespaceAttr(x.GalaxyNamespace, "movie")]; ok {
+			moveOk = true
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+	require.True(t, moveOk)
+	// Setup test directories.
+	dirSetup(t)
+	// Send backup request.
+	//       mostly because of a race condition
+	//       adding sleep
+	time.Sleep(time.Second * 10)
```

Comment:
Why do we need sleep here? Add that to comment as well.

Reply:
Tests run fine without sleep so i removed this code.